### PR TITLE
fix: Delete dir failed when .DS_Store in it

### PR DESCRIPTION
**Pull request by jiuker**
## Description

fix: Delete dir failed when .DS_Store in it
This file will create by macOS use Finder to view it.

## Types of changes
- [x] Bug fix ( #16836  )
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/xl-storage.go [hunk 1]
```diff
@@ -2077,6 +2077,16 @@ func (s *xlStorage) deleteFile(basePath, deletePath string, recursive, force boo
 			if isObjectDir {
 				return errFileNotFound
 			}
+			// if we have .DS_Store from macOS, it will copy to any where
+			storeFilePath := pathJoin(deletePath, ".DS_Store")
```
Comment (klauspost) [inline cmd/xl-storage.go]: Only enable when `if runtime.GOOS == "darwin" {` - this will be pointless in most installs.

**Review by klauspost — CHANGES_REQUESTED** [review]

File: cmd/xl-storage.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/xl-storage.go]: It needs to be changed slightly.

**Review by klauspost — APPROVED** [review]
@harshavardhana Did this fix it for you as well, or did you have another change in mind?

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
> @harshavardhana Did this fix it for you as well, or did you have another change in mind?

This is fine.

**Pull request merged**